### PR TITLE
sync_diff_inspector: show hex data for binary column

### LINKED ISSUE #807: sync-diff-inspector: output binary column in hex form (x'ABCDEF')
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
In mysql:
```
create table tb(a int, b binary(16) primary key);
insert into tb(a,b) values(1,0xBF500C00A2034521B819D6EB7065D200);
```
In tidb:
```
create table tb(a int, b binary(16) primary key);
insert into tb(a,b) values(2,0xBF500C00A2034521B819D6EB7065D200);
```
Now run sync_diff_inspector.

2. What did you expect to see?
```
REPLACE INTO `test`.`tb`(`a`,`b`) VALUES (1,0xBF500C00A2034521B819D6EB7065D200);
```

3. What did you see instead?
```
$ cat test\:tb\:0\:0-0\:0.sql 
-- table: test.tb
-- range in sequence: Full
/*
  DIFF COLUMNS ╏ `A`  
╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╋╍╍╍╍╍╍
  source data  ╏ 1    
╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╋╍╍╍╍╍╍
  target data  ╏ 2    
╍╍╍╍╍╍╍╍╍╍╍╍╍╍╍╋╍╍╍╍╍╍
*/
REPLACE INTO `test`.`tb`(`a`,`b`) VALUES (1,'?P
                                               ?E!???pe?');
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
NA
5. which tool are you using?
NA
6. what versionof tool are you using (`pump -V` or `tidb-lightning -V` or `syncer -V`)?
```
$ sync_diff_inspector -V
App Name: sync_diff_inspector v2.0
Release Version: v6.5.3
Git Commit Hash: 89b804103994cd63273fa9e57e71f270b1a42fcc
Git Branch: heads/refs/tags/v6.5.3
UTC Build Time: 2023-06-05 07:19:58
Go Version: go1.20.3
```

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb-tools/issues/807

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: sync_diff_inspector/utils/utils.go [hunk 1]
```diff
@@ -1043,3 +1043,7 @@ func IsRangeTrivial(rangeCond string) bool {
 	}
 	return strings.ToLower(rangeCond) == "true"
 }
+
+func IsBinaryColumn(col *model.ColumnInfo) bool {
+	return col.GetType() == mysql.TypeString && mysql.HasBinaryFlag(col.GetFlag())
```
Comment (D3Hunter) [inline sync_diff_inspector/utils/utils.go]: does `varbinary` has type `TypeVarString`？

Comment (lance6716) [reply sync_diff_inspector/utils/utils.go]: The unit test should cover this check.

**Review by lance6716 — APPROVED** [review]

**Comment by lance6716** [comment]
/hold

wait to address comment

File: sync_diff_inspector/utils/utils.go (hunk 1 above)
Comment (GMHDBJD) [reply sync_diff_inspector/utils/utils.go]: fix it

**Review by D3Hunter — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**